sway/desktop/xdg_shell.c

```diff
@@ -139,6 +141,16 @@ static struct sway_xdg_shell_view *xdg_shell_view_from_view(
 	return (struct sway_xdg_shell_view *)view;
 }
 
+static const struct wlr_security_context_v1_state *security_context_from_view(
+		struct sway_view *view) {
+	const struct wl_client *client =
+		wl_resource_get_client(view->surface->resource);
```

Comment:
It seems like this has nothing xdg-shell specific - do we really need to get through the `view->impl` abstraction, or can we just put this logic in generic view code?

Reply:
Done.